applications/tari_mining_node/src/stratum/controller.rs

```diff
@@ -77,20 +77,20 @@ impl Controller {
     }
 
     pub fn try_connect(&mut self) -> Result<(), Error> {
-        self.stream = Some(Stream::new());
-        self.stream
-            .as_mut()
-            .unwrap()
-            .try_connect(&self.server_url, self.server_tls_enabled)?;
+        // TODO: Check should we `take` it here?
+        self.stream.take();
```

Comment:
No need to take from the Option. Assigning to it will Drop the previous instance.

Reply:
But if `try_connect` failed do we have to drop the previous connection this way?

Reply:
I don't think it's an issue to connect out but think you're right the disconnect may happen after. Can use take or 
```rust
// Ensure current connection is disconnected
self.stream = None;
```